File: PracticeSolutions/VB.NET/Lesson3/Practice2.vb
Imports ESRI.ArcGIS.ArcMapUI
Imports ESRI.ArcGIS.Carto
Imports ESRI.ArcGIS.Geodatabase
Imports ESRI.ArcGIS.DataSourcesFile

Public Class Practice2
    Inherits ESRI.ArcGIS.Desktop.AddIns.Button

    Public Sub New()

    End Sub

    Public Sub Practice2()

        Dim pMxDoc As IMxDocument
        pMxDoc = My.ArcMap.Application.Document

        Dim pMap As IMap
        pMap = pMxDoc.FocusMap

        Dim pWSFactory As IWorkspaceFactory
        pWSFactory = New ShapefileWorkspaceFactory

        Dim pWorkspace As IWorkspace
        pWorkspace = pWSFactory.OpenFromFile("c:\WCGIS\G5224P\Lesson1_2_3", _
            My.ArcMap.Application.hWnd)

        Dim pFWorkspace As IFeatureWorkspace
        pFWorkspace = pWorkspace 'QI

        Dim pStatesFClass As IFeatureClass
        pStatesFClass = pFWorkspace.OpenFeatureClass("us_boundaries")

        Dim pStatesLayer As IFeatureLayer
        pStatesLayer = New FeatureLayer
        pStatesLayer.FeatureClass = pStatesFClass
        pStatesLayer.Name = "U.S. States"

        pMap.AddLayer(pStatesLayer)

        Dim pCitiesFClass As IFeatureClass
        pCitiesFClass = pFWorkspace.OpenFeatureClass("us_cities")

        Dim pCitiesLayer As IFeatureLayer

        pCitiesLayer = New FeatureLayer
        pCitiesLayer.FeatureClass = pCitiesFClass
        pCitiesLayer.Name = "U.S. Cities"

        pMap.AddLayer(pCitiesLayer)

        Dim pActiveView As IActiveView
        pActiveView = pMap
        pActiveView.Refresh()

        pMxDoc.UpdateContents()

    End Sub

    Protected Overrides Sub OnClick()
        Call Practice2()
    End Sub

    Protected Overrides Sub OnUpdate()

    End Sub
End Class

File: PracticeSolutions/C#/Lesson3/Practice2.cs
using System;
using System.Collections.Generic;
using System.Text;
using System.IO;
using ESRI.ArcGIS.ArcMapUI;
using ESRI.ArcGIS.Carto;
using ESRI.ArcGIS.Geodatabase;
using ESRI.ArcGIS.DataSourcesFile;


namespace Lesson3_Text
{
    public class Practice2 : ESRI.ArcGIS.Desktop.AddIns.Button
    {
        public Practice2()
        {
        }

        public void PracticeExercise2()
                {
            IMxDocument pMxDoc;
            pMxDoc = (IMxDocument)ArcMap.Application.Document;

            IMap pMap;
            pMap = pMxDoc.FocusMap;

            IWorkspaceFactory pWSFactory;
            pWSFactory = new ShapefileWorkspaceFactory();

            IWorkspace pWorkspace;
            pWorkspace = pWSFactory.OpenFromFile("c:\\WCGIS\\G5224P\\Lesson1_2_3", ArcMap.Application.hWnd);
            
            //QI
            IFeatureWorkspace pFWorkspace;
            pFWorkspace = (IFeatureWorkspace)pWorkspace;

            string strFileName = "us_boundaries";
            string strLyrName = "U.S. States";
            ShapefileToLayer(pFWorkspace, pMap, strFileName, strLyrName);

            strFileName = "us_cities";
            strLyrName = "U.S. Cities";
            ShapefileToLayer(pFWorkspace, pMap, strFileName, strLyrName);

            IActiveView pActiveView;
            pActiveView = (IActiveView)pMap;
            pActiveView.Refresh();

            pMxDoc.UpdateContents();
        }

        private void ShapefileToLayer(IFeatureWorkspace pFWorkspace, IMap pMap, string strFileName, string strLyrName)
        {
            IFeatureClass pFClass;
            pFClass = pFWorkspace.OpenFeatureClass(strFileName);

            IFeatureLayer pLayer;
            pLayer = new FeatureLayer();
            pLayer.FeatureClass = pFClass;
            pLayer.Name = strLyrName;

            pMap.AddLayer(pLayer);
        }
        
        protected override void OnClick()
        {
            PracticeExercise2();
        }

        protected override void OnUpdate()
        {
        }
    }
}
